grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy

```diff
@@ -45,51 +46,34 @@ class NewConsentGroupController extends AuthenticatedController {
     }
 
     def save() {
-        Issue consent;
+        Issue consent
+        ConsentCollectionLink consentCollectionLink
         try{
             List<MultipartFile> files = request.multiFileMap.collect { it.value }.flatten()
             User user = getUser()
-            String dataProject = request.parameterMap["dataProject"].toString()
             JsonParser parser = new JsonParser()
-            JsonArray dataProjectJson = parser.parse(dataProject)
+            JsonArray dataProjectJson = parser.parse(request.parameterMap["dataProject"].toString())
+            JsonArray dataConsentCollectionJson = parser.parse(request.parameterMap["dataConsentCollection"].toString())
             Issue issue = IssueUtils.getJson(Issue.class, dataProjectJson[0])
+            consentCollectionLink = IssueUtils.getJson(ConsentCollectionLink.class, dataConsentCollectionJson[0])
             Issue source = queryService.findByKey(issue.getSource())
             if (source != null) {
                 issue.setRequestDate(new Date())
                 consent = issueService.createIssue(IssueType.CONSENT_GROUP, issue)
+                consentCollectionLink.consentKey = consent.projectKey
+                consentCollectionLink.creationDate = new Date()
                 persistenceService.saveEvent(issue.projectKey, user?.displayName, "New Consent Group Added", EventType.SUBMIT_CONSENT_GROUP)
                 try {
-                    // If any sample collections were linked, we need to add them to the consent group.
-                    def sampleCollectionIds = []
-                    if (issue.getSamples()) { sampleCollectionIds.addAll(issue.getSamples()) }
-                    if (sampleCollectionIds.isEmpty()) {
-                        new ConsentCollectionLink(
-                                projectKey: source.projectKey,
-                                consentKey: consent.projectKey,
-                                sampleCollectionId: null,
-                                creationDate: new Date()
-                        ).save(flush: true)
-                    } else {
-                        sampleCollectionIds.each {
-                            new ConsentCollectionLink(
-                                    projectKey: source.projectKey,
-                                    consentKey: consent.projectKey,
-                                    sampleCollectionId: it,
-                                    creationDate: new Date()
-                            ).save(flush: true)
-                        }
-                    }
+                    consentCollectionLink.save(flush: true)
```

Comment:
Nit: save operations should be in a service class ... sorry if I missed this on the first review.